superset-frontend/src/views/CRUD/data/database/types.ts

```diff
@@ -21,6 +21,16 @@ type DatabaseUser = {
   last_name: string;
 };
 
+type DatabaseParameters = {
+  database: string;
+  engine: string;
+  host: string;
+  password?: string;
+  port: number;
+  query: Object;
```

Comment:
I think it is better to use the query props here, which I think you can import.

Reply:
Confusingly, this is a different query: https://docs.sqlalchemy.org/en/14/core/engines.html#add-parameters-to-the-url-query-string

Reply:
oh interesting, so then would Record<string, string> be more explicit here given that it has string keys and values?

Reply:
That sounds like a good plan.

Reply:
I also added in some props in https://github.com/apache/superset/pull/14583. For bigquery most of those props are going be optional (maybe some other dbs too). Are we going to use db parameters for Gsheets for example?

Reply:
@AAfghahi right, `Record<string, string>`

But we need to keep in mind that these are specific for Postgres and similar databases. For other DBs some attributes might be optional, or there might be a different set of attributes, so maybe we should just define the type as `Object`.

Reply:
that's fair, I hadn't considered that.